c7n/resources/ec2.py

```diff
@@ -1698,3 +1698,55 @@ def get_instance_attribute(self, resources, attribute):
             keys.remove('InstanceId')
             resource['c7n:attribute-%s' % attribute] = fetched_attribute[
                 keys[0]]
+
+
+@resources.register('launch-template')
+class LaunchTemplate(query.QueryResourceManager):
+
+    class resource_type(object):
+        service = 'ec2'
+        type = 'LaunchTemplate'
+        id = 'LaunchTemplateId'
```

Comment:
more just fyi, id parameter should generally be how they are the resource filtered/queried/deleted when using the api, afaics this looks fine here because the template api supports both name and ids, a bit further down your filtering by name, but typically we would pass the id to the filter. To properly support serverless execution modes, we'll need to adjust filter_name below to use the `LaunchTemplateIds` parameter filter.

Reply:
added